src/pages/signin/SignInPage.js

```diff
@@ -59,23 +74,37 @@ const defaultProps = {
 /**
  * @param {Boolean} hasLogin
  * @param {Boolean} hasValidateCode
+ * @param {Boolean} hasAccount
  * @param {Boolean} isPrimaryLogin
  * @param {Boolean} isAccountValidated
  * @param {Boolean} hasEmailDeliveryFailure
  * @returns {Object}
  */
-function getRenderOptions({hasLogin, hasValidateCode, hasAccount, isPrimaryLogin, isAccountValidated, hasEmailDeliveryFailure}) {
+function getRenderOptions({hasLogin, hasValidateCode, hasAccount, isPrimaryLogin, isAccountValidated, isSAMLEnabled, isSAMLRequired, isUsingSAMLLogin, hasEmailDeliveryFailure}) {
     const shouldShowLoginForm = !hasLogin && !hasValidateCode;
-    const shouldShowEmailDeliveryFailurePage = hasLogin && hasEmailDeliveryFailure;
+    let shouldShowSAMLEnabledForm = false;
+    let shouldInitiateSAMLLogin = false;
+    const platform = getPlatform();
+
+    // SAML is temporarily restricted to users on the beta or to users signing in on web and mweb
+    if (Permissions.canUseSAML() || platform === CONST.PLATFORM.WEB) {
+        shouldShowSAMLEnabledForm = hasAccount && hasLogin && isSAMLEnabled && !isSAMLRequired && isUsingSAMLLogin;
```

Comment:
Sorry, why ` !SAMLRequired` is here?
Shouldn't we display form if SAML is required?

Reply:
Oh I see, looks like we don't initiate directly and give an option to choose if it is not required

Reply:
Correct!